src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -68,6 +69,17 @@ public override BoundStatement InstrumentFieldOrPropertyInitializer(BoundStateme
             rewritten = base.InstrumentFieldOrPropertyInitializer(original, rewritten);
             SyntaxNode syntax = original.Syntax;
 
+            if (rewritten.Kind == BoundKind.Block)
+            {
+                var block = (BoundBlock)rewritten;
+                return block.Update(block.Locals, block.LocalFunctions, ImmutableArray.Create(InstrumentFieldOrPropertyInitializer(block.Statements.Single(), syntax)));
```

Comment:
Looking at this and your previous PR, I still think your previous PR should just have used `Single()` as well, rather than checking that there is one statement.
Doesn't seem consistent to rely on this assumption only in some places.